identity/src/main/java/com/stripe/android/identity/ml/ModelIO.kt

```diff
@@ -35,13 +36,51 @@ internal sealed interface AnalyzerOutput
 /**
  * Output of IDDetector
  */
-internal data class IDDetectorOutput(
-    val boundingBox: BoundingBox,
-    val category: Category,
-    val resultScore: Float,
-    val allScores: List<Float>,
-    val blurScore: Float
-) : AnalyzerOutput
+internal sealed interface IDDetectorOutput : AnalyzerOutput {
+    class Legacy(
+        val boundingBox: BoundingBox,
+        val category: Category,
+        val resultScore: Float,
+        val allScores: List<Float>,
+        val blurScore: Float
+    ) : IDDetectorOutput
+
+    class Modern(
+        val category: Category,
+        val resultScore: Float,
+        val allScores: List<Float>,
+        val mbOutput: MBDetector.DetectorResult
+    ) : IDDetectorOutput
+
+    fun category(): Category =
+        when (this) {
+            is Legacy -> {
+                this.category
+            }
+            is Modern -> {
+                this.category
+            }
```

Comment:
should these be class properties instead?

Reply:
Yes! moved them to the parent class